services/builder/configProcessor.go

```diff
@@ -216,3 +222,51 @@ func mergeDeleteCounts(dst, src map[string]*nodeCount) map[string]*nodeCount {
 	}
 	return dst
 }
+
+// prepareConfigToDelete function queries for cluster those needs ro be deleted from current state.
+// It also updated the config object to remove the clusters to be deleted from current state.
+// returns *pb.Config which contains clusters (both k8s and lb) that needs to be deleted.
+func prepareConfigToDelete(config *pb.Config) *pb.Config {
```

Comment:
The name `prepareConfigToDelete` should be changed imo. The function is not taking care of the deletion of individual nodes from the node pool, it only checks for the whole cluster deletion. I would prefer if the name would reflect that, i.e. `getDeletedClustersConfig`